### PR TITLE
Added announcements deleter

### PR DESCRIPTION
**Pull request by Revadike**
Resolves https://github.com/tfedor/AugmentedSteam/issues/408

### CODE REVIEW

**Comment by MxtOUT** [comment]
Also not sure about the usefulness

**Comment by tfedor** [comment]
I'm with @MxtOUT here. Not sure how useful this is, moreover it's another bulk action that can't be reversed. Huge potential for error.

**Comment by Revadike** [comment]
> I'm with @MxtOUT here. Not sure how useful this is, moreover it's another bulk action that can't be reversed. Huge potential for error.

Hence the warning before use. It can be used when you were hacked, or a rogue moderator sent a lot of announcements, or if you'd simply clean your history.

**Comment by KarlCastle** [comment]
> > I'm with @MxtOUT here. Not sure how useful this is, moreover it's another bulk action that can't be reversed. Huge potential for error.
> 
> Hence the warning before use. It can be used when your group was hacked, or a rogue moderator sent a lot of announcements, or if you'd simply clean your history.

I've never admin'ed a group. I thought being able to post-announcements also meant you could kick-all-the-other-officers? Who's left to fix the issue?

File: js/content/community.js [hunk 1]
```diff
@@ -3459,6 +3459,128 @@ class WorkshopPageClass {
...
+                    await RequestData.getHttp(url);
+                    i++;
+                    updateWaitDialog();
+                    resolve();
+                });
+            }
+
+            function getAnnouncements(page, hidden) {
+                return new Promise(async resolve => {
+                    let pathSplit = location.pathname.split("/")
```
Comment (KarlCastle) [inline js/content/community.js]: ditto

File: js/content/community.js [hunk 2]
```diff
@@ -3459,6 +3459,128 @@ class WorkshopPageClass {
...
+                        window.dialog.Dismiss();
+                    }
+
+                    window.dialog = ShowBlockingWaitDialog("${modalTitle}", \`${Localization.str.del_announcements_loading.replace("__i__", i).replace("__total__", total)}\`);
+                }`)
+            }
+
+            function deleteAnnouncement(url) {
+                return new Promise(async resolve => {
+                    await RequestData.getHttp(url);
```
Comment (KarlCastle) [inline js/content/community.js]: Why the new promise? and what happens if getHttp rejects?

File: js/content/community.js (hunk 1 above)
Comment (Revadike) [reply js/content/community.js]: ditto?

File: js/content/community.js (hunk 2 above)
Comment (Revadike) [reply js/content/community.js]: uncaught promise rejection

**Comment by Revadike** [comment]
> I've never admin'ed a group. I thought being able to post-announcements also meant you could kick-all-the-other-officers? Who's left to fix the issue (EDIT: if you're "hacked")?

Sure, you can kick officers, after the damage has been done. But you're still left to clean the damage. That's exactly what this feature would be useful for.

File: js/content/community.js (hunk 1 above)
Comment (KarlCastle) [reply js/content/community.js]: "as above": "Why the new promise? and what happens if getHttp rejects?"

You can also refactor out the `new Promise()` or the `async function()` or both here.

File: js/content/community.js (hunk 2 above)
Comment (KarlCastle) [reply js/content/community.js]: From the user's perspective, they don't see the console red text. So this would be "failing silently".

**Comment by tfedor** [comment]
This is also making irreversible bulk changes and also has a potential for misuse. I don't think this feature would be used often enough to grant the need to implement it in AS.

**Pull request closed without merging**